docs/dev/architeture.md

```diff
@@ -0,0 +1,206 @@
+- The information of `SST`, such as `SST` path.
+
+Now the `Manifest` is based on `WAL` (this is a different instance from the `WAL` mentioned above for newly written data) and in order to avoid infinite expansion of metadata (actually every `Flush` leads to an update on sst information), `Snapshot` is also introduced to clean up the history of metadata updates.
+
+#### Object Store
+module path: https://github.com/CeresDB/ceresdb/tree/main/components/object_store
+
+The `SST` generated by `Flush` needs to be persisted and the abstraction of the persistent storage device is `ObjectStore` including multiple implementations:
+- Based on local file system;
+- Based on [Alibaba Cloud OSS](https://www.alibabacloud.com/product/object-storage-service).
+
+The distributed architecture of CeresDB separates storage and computing, which requires `Object Store` needs to be a highly available and reliable service independent of CeresDB. Therefore, storage systems like [Alibaba Cloud OSS](https://www.alibabacloud.com/product/object-storage-service) is a good choice and in the future implementations on storage systems of some other cloud service providers is planned to provide.
+
+#### SST
+module path: https://github.com/CeresDB/ceresdb/tree/main/analytic_engine/src/sst
+
+Both `Flush` and `Compaction` involves `SST` and in the codebase `SST` itself is actually an abstraction that can have multiple specific implementations. The current implementation is based on [Parquet](https://parquet.apache.org/), which is a column-oriented data file format designed for efficient data storage and retrieval.
+
+The format of `SST` is very critical for retrieving data and is also the most important part to perform well in handling both timeseries and analytic workloads. At present, our [Parquet](https://parquet.apache.org/)-based implementation is good at processing analytic workload but is poor at processing timeseries workload. In our road map, a new hybrid storage format is planned to be able to process the two workloads at the same time.
+
+#### Space
+module path: https://github.com/CeresDB/ceresdb/blob/main/analytic_engine/src/space.rs
+
+In `Analytic Engine`, there is a concept called `space` and here is an explanation for it to resolve some ambiguities when read source code. Actually `Analytic Engine` does not have the concept of `catalog` and `schema` and only provides two levels of relationship: `space` and `table`. And in the implementation, the `schema id` (which should be unique across all `catalog`s) on the upper layer is actually mapped to `space id`.
+
+The `space` in `Analytic Engine` serves mainly for isolation of resources for different tenants, such as the usage of memory.
+
+## Critical Path
+After a brief introduction to some important modules of CeresDB, we will give a description for some critical paths in code, hoping to provide interested developers with a guide for reading the code.
+
+### Query
```

Comment:
It would be better if we provide a flowchart to explain the query/write process.

- https://plantuml.com/ascii-art